### ISSUE TITLE
Issue Reproducing antmaze-medium-play-v2 Results

### ISSUE THREAD

**Issue by fangqi-Zhu** [MaxSobolMark/PolicyAgnosticRL#3]
Thank you for your excellent work. I'm encountering an issue while trying to reproduce the results for antmaze-medium-play-v2.

Specifically, I've found that pretraining with DDPM shows no performance improvement. However, using PARL-CALQL does yield some performance, but it starts from around 0.3 instead of the expected 0.8 or higher.

Could you please provide some insights into what might be causing this discrepancy?

Results for offline pretraining

`XLA_PYTHON_CLIENT_PREALLOCATE=false env -u PYOPENGL_PLATFORM python ./train.py --environment_name=antmaze-medium-play-v2 --wandb_experiment_name=ddpm_bc_antmaze-v2 --config=./configs/state_config.py:ddpm --num_offline_epochs=3000 --num_online_epochs=0 --seed=0`

![Image](https://github.com/user-attachments/assets/3affa0fd-172e-4117-9520-eaf15ab0f10b)

Results for online fine-tuning

`XLA_PYTHON_CLIENT_PREALLOCATE=false env -u PYOPENGL_PLATFORM python ./train.py --environment_name=antmaze-medium-play-v2 --wandb_experiment_name=parl_antmaze-medium-antmaze-v2 --config=./configs/state_config.py:parl_calql --seed=0 --reward_bias=-1 --config.agent_kwargs.critic_network_kwargs.hidden_dims=256,256,256,256 --config.base_policy_path=ddpm:./results/PA-RL/ddpm_bc_antmaze-v2/seed_0/agent_checkpoints/checkpoint_3000 --config.agent_kwargs.cql_alpha=0.005 --num_offline_epochs=1000 --num_online_epochs=1000 --config.agent_kwargs.distributional_critic_kwargs.q_min=-100 --config.agent_kwargs.critic_ensemble_size=10 --config.data_collection_particle_choosing_strategy="max_q_value" --config.evaluation_particle_choosing_strategy="max_q_value"
`
![Image](https://github.com/user-attachments/assets/8bce79d2-50c6-4613-ba84-0ac5a86162a4)

**Comment by MaxSobolMark** [comment]
Hello!

The first script only pre-trains the diffusion policy with supervised learning - the first half of the pre-training stage. The poor performance here is to be expected. The `antmaze` datasets contain many bad actions, so supervised learning behavior will not be good.

The second script does not just run online fine-tuning. It actually starts by offline pre-training of the critic. Note the `--num_offline_epochs=1000` parameter. The performance until step 1000 will be fully offline, so this looks in line with the paper results!

**Comment by fangqi-Zhu** [comment]
> Hello!
> 
> The first script only pre-trains the diffusion policy with supervised learning - the first half of the pre-training stage. The poor performance here is to be expected. The `antmaze` datasets contain many bad actions, so supervised learning behavior will not be good.
> 
> The second script does not just run online fine-tuning. It actually starts by offline pre-training of the critic. Note the `--num_offline_epochs=1000` parameter. The performance until step 1000 will be fully offline, so this looks in line with the paper results!

Thank you for your prompt response.

Does this imply that by solely pre-training the critic and then, during testing, sampling 32 actions based on a diffusion policy with zero initial performance, followed by global and local optimization, it's possible to achieve near-perfect performance? I ask because I observed that the diffusion policy is not updated during the first 1000 epoch; only the critic is updated. I would like to confirm if my understanding is correct.

![Image](https://github.com/user-attachments/assets/732964c3-1cae-4371-a6dc-bf4af067d3fb)

**Comment by MaxSobolMark** [comment]
Yes, this is exactly correct! This is because `antmaze` domains have high coverage of actions and states, which is not true for other tasks/datasets.
